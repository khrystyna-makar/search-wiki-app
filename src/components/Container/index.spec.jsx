import { mount } from "enzyme";
import Container from '.'
import { beforeEach, describe, expect, it, vi } from "vitest";

vi.mock('../../hooks', () => ({
    useDebounce: vi.fn(() => ({})),
    useSearch: vi.fn(() => ({articles: []})),
    useSearchForm: vi.fn(() => ({
        searchValue: 'test',
        onSearchChange: vi.fn()
    }))
}));

const render = (props, children) => mount(<Container {...props}>{children}</Container>);

const Children = () => <div>test</div>

describe('Container component', () => {
    let sut;
    let props;
    let children = ({searchValue, onSearchChange, articles}) => <Children searchValue={searchValue} onSearchChange={onSearchChange} articles={articles} />;

    describe('without props', () => {
        beforeEach(() => {
            sut = render(props, children);
        });
        it('should match snapshot', () => {
            expect(sut).toMatchSnapshot();
        });
    });

    describe('when passed custom props', () => {
        beforeEach(() => {
            sut = render(props, children);
        });
        it('should return searchValue in Children component', () => {
            const { searchValue } = sut.find('Children').props();
            expect(searchValue).toBe('test')
        });
    })
})