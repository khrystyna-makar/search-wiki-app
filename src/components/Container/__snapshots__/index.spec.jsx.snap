// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Container component > without props > should match snapshot 1`] = `ReactWrapper {}`;
